### DISCUSSION TITLE
Mini-Plus WiFi setup not working (for me). Manual setup in a file possible?

### DISCUSSION THREAD

**Discussion by IsNotAvaliableIsNotAvaliable** [OnionUI/Onion#1490] [Support] [answered]
Hey guys!

Got issues using the WLAN option, get always bad password or something. So my idea is to manualy set  it up in a config file i can't find. I was able to set the WLAN up temp. using the consol app and wpa_supplicant. But this was half an hour of "typing" so it is not a real solution if i want to stay sane :D

Is there any way to setup networking by hand? I'm also willing to fix this in Onion OS for good and contribute some code, but got no idea where in the code the networking-setup is done. Any directions would be nice.

Thanks!

**Comment by FDrebin** [comment]
Are you running into it due to a special character in your password? https://github.com/OnionUI/Onion/issues/780

**Comment by thomasklemenc** [comment, marked as answer]
You can try this for now:

https://github.com/OnionUI/Onion/issues/780#issuecomment-1551799480

or @XK9274 's [better-wifi-miyoo](https://github.com/XK9274/better-wifi-miyoo)

In the future, we'll have the integrated WiFi menu replaced with our own with support for special characters, hidden networks etc. 

A lot of this already done in [this branch](https://github.com/OnionUI/Onion/tree/feat-replace-settings-with-tweaks) but it still needs some work.

**Comment by IsNotAvaliableIsNotAvaliable** [comment]
Thanks guys!

Yes, my Wifi password is very long and includes all sorts of chars outside 0-9a-z. I wonder what the original creator had in mind if he wrote this code that does not support special chars for Wifi passwords? :D

Got it working now using https://github.com/XK9274/better-wifi-miyoo

Glad to hear, that you guys already work on it. Wifi on that little bugger is not that important but nice to have anyways. Great project, keep up the good work!
